ministryofjustice/analytics-platform-cran-proxy
# Fat Package support

This attempts to make binary packages portable over small differences in the OS
execution environment. We have some images based on rocker:shiny and others on
rocker:rstudio and sometimes the rstudio image won't have a library that the
shiny one does. This bundles binary dependencies (but excludes common linux
ones) with the cached R package

This is inspired by:

* [skinnywhale](https://github.com/djosephsen/skinnywhale)
* [conda's way of providing shared
  libraries](https://conda.io/docs/user-guide/tasks/build-packages/use-shared-libraries.html)
* [nixos](https://nixos.wiki/wiki/Packaging/Binaries)


## How to test
1. use rstudio to install a package using r-studio, for example `TTR` using the cran-proxy in alpha (currently running master version): 
    * `> install.packages('TTR', repos = 'https://cran-proxy.services.alpha.mojanalytics.xyz')`
    * make sure you got the binary package not the source package, if you got a source package then wait a minute and try again, you should get a binary one.
    * Try to load the library with `library(TTR)` - you should get an error about not being able to open a shared library.

2. repeat the above with the cran-proxy running on dev (this version of the code):
    * `> install.packages('TTR', repos = 'https://cran-proxy.services.dev.mojanalytics.xyz')`
    * Try to load the library with `library(TTR)` - you should get a warning about the binary being compiled on R3.4.3 but it should load fine (no error)
